core/src/main/scala/org/apache/spark/util/ThreadStackTrace.scala

```diff
@@ -24,4 +24,9 @@ private[spark] case class ThreadStackTrace(
   threadId: Long,
   threadName: String,
   threadState: Thread.State,
-  stackTrace: String)
+  stackTrace: String,
+  blockedByThreadId: Long,
```

Comment:
How about using Option instead of negative/null values? not much difference either way. I'm not sure if thread IDs can be negative ever.